### PR TITLE
Updater changes needed by TAF

### PR DESCRIPTION
**Pull request by renatav**
Changes needed to support development of our updater.

### CODE REVIEW

File: tuf/client/updater.py [hunk 1]
```diff
@@ -2334,14 +2330,15 @@ def _ensure_not_expired(self, metadata_object, metadata_rolename):
     # '1985-10-21T01:22:00Z'.)  Convert it to a unix timestamp and compare it
     # against the current time.time() (also in Unix/POSIX time format, although
     # with microseconds attached.)
-    current_time = int(time.time())
+
+    time = self.update_handler.earliest_valid_expiration_time(metadata_rolename)
```
Comment (danixeee) [inline tuf/client/updater.py]: `Redefining name 'time' from outer scope (line 127)pylint(redefined-outer-name)`

Unused import `time` on line 127, but I would rename this variable to `earliest_exp_time` or some similar, more meaningful, name.

File: tuf/client/handlers.py [hunk 2]
```diff
@@ -35,7 +37,15 @@ def __init__(self, mirrors, repository_directory):
     self.mirrors = mirrors
     self.repository_directory = repository_directory
 
+  def earliest_valid_expiration_time(self, metadata_rolename):
```
Comment (danixeee) [inline tuf/client/handlers.py]: If `metadata_rolename` is not always needed,I would set default `None` value

File: tuf/client/handlers.py [hunk 3]
```diff
@@ -35,7 +37,15 @@ def __init__(self, mirrors, repository_directory):
...
+    # pylint: disable=unused-argument
+    # metadata role name is not needed here, but the git updater
+    # does need it
+    return int(time.time())
 
+  def ensure_not_changed(self, metadata_filename):
+    # not necessary in all cases
```
Comment (danixeee) [inline tuf/client/handlers.py]: docstring will be helpful here

File: tuf/client/handlers.py [hunk 4]
```diff
@@ -92,64 +99,26 @@ class RemoteMetadataUpdater(MetadataUpdater):
...
-      None.
-
-    <Returns>
-      A list of mirrors from which the specified file can be downloaded.
-    """
-    return tuf.mirrors.get_list_of_mirrors('meta', remote_filename,
-      self.mirrors)
-
+  def get_mirrors(self, **kwargs):
+    file_type = kwargs['file_type']
```
Comment (danixeee) [inline tuf/client/handlers.py]: `kwargs.pop('file_type', None)` to avoid `KeyError`. Same for line 104.

File: tuf/client/handlers.py [hunk 5]
```diff
@@ -92,64 +99,26 @@ class RemoteMetadataUpdater(MetadataUpdater):
...
-    Side Effects>
-      A 'securesystemslib.util.TempFile' object is created on disk to store the
-      contents of 'url'.
-
-    <Returns>
-      A 'securesystemslib.util.TempFile' file-like object that points to the
-      contents of 'url'.
-    """
+  def get_metadata_file(self, file_mirror, **kwargs):
+    upperbound_filelength = kwargs['upperbound_filelength']
```
Comment (danixeee) [inline tuf/client/handlers.py]: `kwargs.pop('upperbound_filelength', None)`

File: tuf/client/handlers.py [hunk 6]
```diff
@@ -92,64 +99,26 @@ class RemoteMetadataUpdater(MetadataUpdater):
...
     return tuf.download.unsafe_download(file_mirror,
-        _upperbound_filelength)
+        upperbound_filelength)
+
+  def get_target_file(self, file_mirror, **kwargs):
+    download_safely = kwargs['download_safely']
```
Comment (danixeee) [inline tuf/client/handlers.py]: `kwargs.pop('download_safely', True)` if not passed? Line 115. should also use `pop`

Comment (renatav) [reply tuf/client/handlers.py]: Resolved all

**Review by danixeee — APPROVED** [review]
LGTM; merging

**Pull request merged**